### DISCUSSION TITLE
Particles equality with other objects

### DISCUSSION THREAD

**Discussion by Quettle** [PlasmaPy/PlasmaPy#1221] [Particles] [answered]
Hi, 
I ran into some issues wile using a list of both Particles and CustomParticles, as Particle.__eq__ returns an error if the type of the object to be compared with is not Particle or str. 
Wouldn't it be more convenient if the equality test returned False instead of an error, like with most python types?
For example I would like to be able to do something like `particles.CustomParticle(4*u.u,1*u.C) in [particles.electron, particles.Particle("He"), particles.CustomParticle(4*u.u,1*u.C)]`

What's your opinion ?

**Comment by rocco8773** [comment, marked as answer]
@Quettle Thanks for posting your question!  Yeah, I agree with you that it makes sense for `False` to be returned in these cases.  @namurphy  Was there any rational for the current behavior when `CustomParticle` and `Particle` were initially developed?

**Reply by namurphy** [reply]
My rationale was probably along the lines of...this comparison is invalid, so it should raise an exception.  However, I agree it makes more sense to change this behavior to have `Particle.__eq__` return `False` when the other object is not a `Particle` or something that can be converted to a `Particle`, since that's more in line with how other objects in Python behave (e.g., `1 == "1"` returns `False`).  Thank you for bringing this up!

**Comment by rocco8773** [comment]
@Quettle I saw that you recently did PR #1216, would you be willing to take this on too?

**Reply by Quettle** [reply]
Sure !
